cmake/linker/ld/target.cmake

```
@@ -92,7 +92,7 @@ function(toolchain_ld_link_elf)
     ${LINKERFLAGPREFIX},--whole-archive
     ${ZEPHYR_LIBS_PROPERTY}
     ${LINKERFLAGPREFIX},--no-whole-archive
-    kernel
+    ${ZEPHYR_NO_WHOLE_ARCHIVE_LIBS_PROPERTY}
```

please keep original implementation.
```suggestion
    kernel
```